enforcer/src/main/java/org/wso2/micro/gateway/enforcer/api/API.java

```diff
@@ -38,7 +39,14 @@
 
     default boolean executeFilterChain(RequestContext requestContext) {
         boolean proceed;
-        for (Filter filter : getFilters()) {
+        List<Filter> filterList = getFilters();
+        String lifeCycleState = requestContext.getMathedAPI().getAPIConfig().getApiLifeCycleState();
+        if (lifeCycleState.equals(APIConstants.PROTOTYPED_LIFE_CYCLE_STATUS)) {
```

Comment:
What would happen if the API is there, but lifecycle state is CREATED?

Reply:
When lifecycle state of an API is in CREATED state, that API has not been displayed at devportal. hence  it couldn't be exposed outsider to do the invocations.

Reply:
yeah, You are right.